### ISSUE TITLE
TLS stream with MONO

### ISSUE THREAD

**Issue by JTrotta** [chkr1011/MQTTnet#56]
I found an issue using TLS and MONO during connection. 
To allow secured connections to work in MONO, we need to manage the RemoteCertificateValidationCallback of the SSLStream. So into MqttTcpChannel :

`_sslStream = new SslStream(new NetworkStream(_socket, true), false, this.userCertificateValidationCallback, this.userCertificateSelectionCallback);`

The two callbaack should be exposed to the calling application and the userCertificateValidationCallback should be assigned to a function returning true:

`        private bool remoteValidation(object sender, 
            X509Certificate certificate, 
            X509Chain chain, 
            System.Net.Security.SslPolicyErrors sslPolicyErrors)
        {
            return true;
        }`

this way we can use the library with TLS in MONO.

JJ

**Comment by chkr1011** [comment]
Please have a look at ticket #30. This is this feature already discussed. The problem is that the event cannot be used because it is not available for every platform. In UWP there is no such event. There is only an enum (Flags) which can be set first. So I have to implement a similar enum too and map the behavior to every platform which is currently supported. Internally the mentioned event can be used.

**Comment by JTrotta** [comment]
Good solution.
in the meanwhile I have patched the MQTTnet.NetFramework only.
JJ

**Comment by chkr1011** [comment]
Please have a look at the develop branch. I added several TLS options regarding certificate validation. Please let me know if this fits your needs.

**Comment by JTrotta** [comment]
It works. But I would have preferred a function parameter to be used or not, and not the static function.

Thank you.
JJ

**Comment by chkr1011** [comment]
I know this is a ugly solution. The problem is that the .net framework has different API for this in UWP and thus I cannot just pass a parameter etc. I will consider adding this to the factory which is unique per platform (Factory constructor). Then it is at least no more static. 

Best regards
Christian

**Comment by JTrotta** [comment]
ManagedClient lost this feature, may I add it again? Into tlsoption would be better.

**Comment by chkr1011** [comment]
Yes it is a static function because the implementations are very different between netstandard and UWP. I was not able to define a wrapper which matches both. So a static event is (currently) the best solution.
